corehq/blobs/tests/test_migrate.py

```diff
@@ -434,31 +440,147 @@ def modify(item):
-            doc = SavedBasicExport(configuration=_mk_config("config-%s" % i))
-            doc.save()
-            doc.set_payload(("content %s" % i).encode('utf-8'))
-            docs.append(doc)
+        lost_db = TemporaryFilesystemBlobDB() # must be created before S3 dbs
+        db1 = TemporaryS3BlobDB(config)
+        assert get_blob_db() is db1, (get_blob_db(), db1)
+        missing = "found.not"
+        name = "blob.bin"
+        data = b'binary data not valid utf-8 \xe4\x94'
 
-        s3db = TemporaryS3BlobDB(config)
-        self.db = TemporaryMigratingBlobDB(s3db, fsdb)
+        self.not_founds = set()
+        self.couch_docs = []
+        with lost_db:
+            for doc_type, model_class in self.couch_doc_types.items():
+                item = model_class()
+                item.doc_type = doc_type
+                item.save()
+                item.put_attachment(data, name)
+                with install_blob_db(lost_db):
+                    item.put_attachment(data, missing)
+                    self.not_founds.add((
+                        doc_type,
+                        item._id,
+                        item.external_blobs[missing].id,
+                        item._blobdb_bucket(),
+                    ))
+                item.save()
+                self.couch_docs.append(item)
+
+        def create_obj(rex):
+            ident = random_url_id(8)
+            args = {rex.blob_helper.id_attr: ident}
+            fields = {getattr(f, "attname", "")
+                for f in rex.model_class._meta.get_fields()}
+            if "content_length" in fields:
+                args["content_length"] = len(data)
+            elif "length" in fields:
+                args["length"] = len(data)
+            item = rex.model_class(**args)
+            save_attr = rex.model_class.__name__ + "_save"
+            if hasattr(self, save_attr):
+                getattr(self, save_attr)(item, rex)
+            else:
+                item.save()
+            return item, ident
+        self.sql_docs = []
+        with mod.allow_form_processing_queries():
+            for rex in self.sql_reindex_accessors:
+                item, ident = create_obj(rex)
+                helper = rex.blob_helper({"_obj_not_json": item})
+                db1.put(StringIO(data), ident, helper._blobdb_bucket())
+                self.sql_docs.append(item)
+                lost, lost_blob_id = create_obj(rex)
+                self.not_founds.add((
+                    rex.model_class.__name__,
+                    lost.id,
+                    lost_blob_id,
+                    rex.blob_helper({"_obj_not_json": lost})._blobdb_bucket(),
+                ))
+
+        self.test_size = len(self.couch_docs) + len(self.sql_docs)
+        db2 = TemporaryS3BlobDB(config)
+        self.db = TemporaryMigratingBlobDB(db2, db1)
         assert get_blob_db() is self.db, (get_blob_db(), self.db)
         BaseMigrationTest.discard_migration_state(self.slug)
```

Comment:
Most of it could be moved `setUpClass`, but since it's  only one test...

Agree on the long and hard to follow comment, and I'm not happy with how much code was needed to setup for this test.